src/wallet/rpcwallet.cpp

```diff
@@ -553,6 +553,45 @@ UniValue listaddressgroupings(const JSONRPCRequest& request)
     return jsonGroupings;
 }
 
+UniValue listaddressbalances(const JSONRPCRequest& request)
+{
+    if (!EnsureWalletIsAvailable(request.fHelp))
+        return NullUniValue;
+
+    if (request.fHelp || request.params.size() > 1)
+        throw std::runtime_error(
+            "listaddressbalances ( minamount )\n"
+            "\nLists addresses of this wallet and their balances\n"
+            "\nArguments:\n"
+            "1. minamount               (numeric, optional, default=0) Minimum balance in " + CURRENCY_UNIT + " an address should have to be shown in the list\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"address\": amount,       (string) The dash address and the amount in " + CURRENCY_UNIT + "\n"
+            "  ,...\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("listaddressbalances", "")
+            + HelpExampleCli("listaddressbalances", "10")
+            + HelpExampleRpc("listaddressbalances", "")
+            + HelpExampleRpc("listaddressbalances", "10")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    // Whether to include addresses with zero balances
+    CAmount nMinAmount = 0;
+    if (request.params.size() > 0)
+        nMinAmount = request.params[0].get_int64() * COIN;
```

Comment:
This fails with `JSON integer out of range (code -1)` for non-integer values of the `minamount` parameter. Not sure if `get_int64()` is what we want here since any real => a single duff (`1/COIN`) is a valid balance :thinking:

Reply:
True. The current behavior just seemed a little confusing - `listaddressbalances 10` works, but `listaddressbalances 10.1` fails. A few other RPCs that accept a parameter in DASH use `AmountFromValue(param)` - see [SendToAddress](https://github.com/dashpay/dash/blob/develop/src/wallet/rpcwallet.cpp#L418). Perhaps that would be useful here...

Reply:
Agree, `AmountFromValue` is normally used to support `10.1` style amounts. Missed this while reviewing.

Reply:
my bad 🙈 will fix